### PR TITLE
Feature/FC-1210

### PR DESCRIPTION
**Pull request by chrbrt**
- added `lpf metrics` command which allows displaying and downloading function metrics as json or csv

### CODE REVIEW

File: src/controller/metrics.controller.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    if (startTimestamp > endTimestamp) {
+      throw new Error('Start timestamp has to be before end timestamp.');
+    }
+
+    if (endTimestamp - startTimestamp > THIRTY_DAYS) {
+      throw new Error('Time period cannot exceed 30 days.');
+    }
+
+    if (endTimestamp - Number(startTimestamp) < FIFTEEN_MINUTES) {
```
Comment (Templum) [inline src/controller/metrics.controller.ts:72]: Why only here the Number casting? Seems 🐟

File: src/service/faas.service.ts [hunk 2]
```diff
@@ -302,6 +302,44 @@ export class FaasService implements IFaaSService {
...
+  //   bucketSize: any;
+  // }) {
+  //   const url = `/reports/invocations`;
+  //   const additionalParams = `&startTimestamp=${startTimestamp}&endTimestamp=${endTimestamp}&bucketSize=${bucketSize}&invocationStates=UNKOWN&invocationStates=SUCCEEDED&invocationStates=CODING_FAILURE&invocationStates=PLATFORM_FAILURE&invocationStates=TIMEOUT`;
+  //   return this.doFetch({
+  //     urlPart: url,
+  //     additionalParams,
+  //     method: 'GET',
+  //   });
+  // }
```
Comment (Templum) [inline src/service/faas.service.ts:321]: If unused should be removed. We should not include 💀 code

File: src/service/faas.service.ts [hunk 3]
```diff
@@ -302,6 +302,44 @@ export class FaasService implements IFaaSService {
...
+    bucketSize,
+  }: {
+    uuid: string;
+    startTimestamp: number;
+    endTimestamp: number;
+    bucketSize: any;
+  }) {
+    const url = `/reports/invocations/${uuid}`;
+    const additionalParams = `&startTimestamp=${startTimestamp}&endTimestamp=${endTimestamp}&bucketSize=${bucketSize}&invocationStates=UNKOWN&invocationStates=SUCCEEDED&invocationStates=CODING_FAILURE&invocationStates=PLATFORM_FAILURE&invocationStates=TIMEOUT`;
+    return this.doFetch({
```
Comment (Templum) [inline src/service/faas.service.ts:336]: Are the tests still missing ? If so we should include at least one

**Review by Templum — CHANGES_REQUESTED** [review]

File: src/controller/metrics.controller.ts (hunk 1 above)
Comment (chrbrt) [reply src/controller/metrics.controller.ts:72]: Forgot to remove. Validity is checked before

**Review by jplieninger — APPROVED** [review]

**Review by Templum — APPROVED** [review]
LGTM

**Review by Templum — APPROVED** [review]
LGTM

**Pull request merged**